src/services/download/gcs.rs

```diff
@@ -118,41 +111,45 @@ fn get_auth_jwt(source_key: &GcsSourceKey, expiration: i64) -> Result<String, Gc
 #[derive(Debug)]
 pub struct GcsDownloader {
     token_cache: Mutex<GcsTokenCache>,
+    client: reqwest::Client,
 }
 
 impl GcsDownloader {
     pub fn new() -> Self {
         let token_cache = Mutex::new(GcsTokenCache::new(GCS_TOKEN_CACHE_SIZE));
-        Self { token_cache }
+        let client = reqwest::Client::new();
+        Self {
+            token_cache,
+            client,
+        }
     }
 
     /// Requests a new GCS OAuth token.
     async fn request_new_token(&self, source_key: &GcsSourceKey) -> Result<GcsToken, GcsError> {
         let expires_at = Utc::now() + Duration::minutes(58);
         let auth_jwt = get_auth_jwt(source_key, expires_at.timestamp() + 30)?;
 
-        let mut builder = client::post("https://www.googleapis.com/oauth2/v4/token");
-        // for some inexplicable reason we otherwise get gzipped data back that actix-web
-        // client has no idea what to do with.
-        builder.header("accept-encoding", "identity");
-        let response = builder
+        let request = self
+            .client
+            .post("https://www.googleapis.com/oauth2/v4/token")
+            // TODO(ja): check if this still holds
```

Comment:
I think encodings are feature flags on reqwest, or they are builtin even?

Reply:
According to the reqwest main readme those are off-by-default feature flags, correct. This PR is still very much in progress and needed some leg work to be merged first, so all of this still needs some time to get right :)